internal/controllers/topology/cluster/structuredmerge/dryrun.go

```diff
@@ -52,16 +53,46 @@ func dryRunSSAPatch(ctx context.Context, dryRunCtx *dryRunSSAPatchInput) (bool,
 	}
 
```

Comment:
I'll try to add a unit test for the dryrun func with KubeadmConfigTemplate & envtest (to cover / reproduce exactly the issue we had)

Have to see how complicated it will be though